include/libdnf/base/goal_elements.hpp

```diff
@@ -137,26 +137,26 @@ struct GoalJobSettings : public ResolveSpecSettings {
     /// Return used value for clean_requirements_on_remove
     GoalUsedSetting get_used_clean_requirements_on_remove() const { return used_clean_requirements_on_remove; };
 
+    /// Optionally assign AdvisoryQuery that is used to filter goal target packages (used for upgrade and install)
+    void set_advisory_filter(const libdnf::advisory::AdvisoryQuery & filter) { advisory_filter = filter; };
+
+    // Which types of group packages are going to be installed with the group.
+    // If not set, default is taken from ConfigMain.group_package_types
+    void set_group_package_types(const libdnf::comps::PackageType type) { group_package_types = type; }
+
     /// Set whether hints should be reported
     bool report_hint{true};
-    /// Set strict, AUTO means that it is handled according the the default behavior
+    /// Set strict, AUTO means that it is handled according to the default behavior
     GoalSetting strict{GoalSetting::AUTO};
-    /// Set best, AUTO means that it is handled according the the default behavior
+    /// Set best, AUTO means that it is handled according to the default behavior
     GoalSetting best{GoalSetting::AUTO};
-    /// Set clean_requirements_on_remove, AUTO means that it is handled according the the default behavior
+    /// Set clean_requirements_on_remove, AUTO means that it is handled according to the default behavior
     GoalSetting clean_requirements_on_remove{GoalSetting::AUTO};
-    /// Define which installed packagies should be modified acording to repoid from which they were installed
+    /// Define which installed packages should be modified according to repoid from which they were installed
     std::vector<std::string> from_repo_ids;
     /// Reduce candidates for the operation according repository ids
     std::vector<std::string> to_repo_ids;
 
-    /// Optionally assign AdvisoryQuery that is used to filter goal target packages (used for upgrade and install)
-    std::optional<libdnf::advisory::AdvisoryQuery> advisory_filter;
```

Comment:
Right now we don't need a getter, but I think it would be logical to have it. GoalJobSetting is stored in event log to provide all information related to the job request, therefore there could be a reason to provide a getter. But what could the getter return?

Reply:
The problem in this case is also that the settings object is being copied to the log, so this needs to be beard in mind and f.e. `unique_ptr` approach is not possible there.

Reply:
In the case we want to have a getter here we could use the similar approach as in the commit for `transaction` changes - return raw pointer which would either return pointer to data if assigned, or `nullptr`.

Reply:
Raw pointer is an option that we agreed on.